### PR TITLE
Redeem & Unstake integration test with proof

### LINKED ISSUE #614: Redeem and Unstake integration test (proof)

### PR DESCRIPTION
**Pull request by abhayks1**
Fixes #614 

PR takes care of redeem and unstake integration test with proof.

### CODE REVIEW

**Comment by deepesh-kn** [comment]
Reviewing 🦊

File: test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+    true,
+  );
+};
+
+/*
+ * Performs Gateway.progressUnstakeWithProof.
+ *
+ * @return {Promise<void>}
+ */
+const progressUnstakeWithProof = async (messageStatus) => {
```
Comment (deepesh-kn) [inline test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js]: Similarly, update the documentation to include param `messageStatus `

File: test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+  const event = EventDecoder.getEvents(tx, gateway);
+  ConfirmRedeemIntentAssertion.verify(event, redeemRequest);
+};
+
+/*
+ * Performs CoGateway.progressRedeemWithProof.
+ *
+ * @return {Promise<void>}
+ */
+const progressRedeemWithProof = async (messageStatus) => {
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js]: Done!

File: test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js (hunk 1 above)
Comment (abhayks1) [reply test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js]: Done!

File: test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+    progressUnstakeAssertion = new ProgressUnstakeAssertion(
+      gateway,
+      token,
+      baseToken,
+    );
+    proofUtils = new ProofUtils(auxiliaryWeb3, originWeb3);
+    messageBoxOffset = await cogateway.MESSAGE_BOX_OFFSET.call();
+  });
+
+  it('Redeem & Unstake (progress redeem and then progress unstake', async () => {
```
Comment (jasonklein) [inline test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js]:
```suggestion
  it('Redeem & Unstake (progress redeem and then progress unstake)', async () => {
```

File: test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+    await redeem();
+    await confirmRedeemIntent();
+    await progressRedeemWithProof(MessageStatus.Declared);
+    await progressUnstakeWithProof(MessageStatus.Progressed);
+  });
+
+  it('Redeem & Unstake (progress unstake and then progress redeem', async () => {
```
Comment (jasonklein) [inline test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js]:
```suggestion
  it('Redeem & Unstake (progress unstake and then progress redeem)', async () => {
```

**Comment by abhayks1** [comment]
> Also, please point the `fixes` keyword in this PR to #614 and not #638.

Ahh 😞. Thank you for pointing out 💯

File: test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js (hunk 3 above)
Comment (abhayks1) [reply test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js]: Done!

File: test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js (hunk 4 above)
Comment (abhayks1) [reply test_integration/03_redeem_and_unstake/02_redeem_and_unstake_proof.js]: Done!

**Pull request merged**